### PR TITLE
Fix types to support Ajv plugins with options

### PR DESCRIPTION
**Pull request by pverdile**
Hi, maintainers,
I couldn't work with ajv plugins options as described in documentation [Server:ajv](https://www.fastify.io/docs/latest/Reference/Server/#ajv) in a typescript file, so I realized that types were incomplete.

I add the necessary types and their tests.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: fastify.d.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ export type FastifyServerOptions<
   return503OnClosing?: boolean,
   ajv?: {
     customOptions?: AjvOptions,
-    plugins?: Function[]
+    plugins?: (Function | [Function, string | string[]])[]
```
Comment (climba03003) [inline fastify.d.ts:147]: It is the plugin options and we should not expect it to be `string` or `string[]` only.
```suggestion
    plugins?: (Function | [Function, unknown])[]
```

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: fastify.d.ts (hunk 1 above)
Comment (pverdile) [reply fastify.d.ts:147]: I totally agree, thanks.

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Comment by Eomm** [comment]
ping @climba03003

**Review by climba03003 — APPROVED** [review]
LGTM.

**Pull request merged**